examples/plot_replay_experiment.py

```diff
@@ -0,0 +1,149 @@
+To achieve that, cardinal allows to store intermediate variables,
+such as selected samples, in a cache. Users can therefore resume
+an interrupted experiment, using the ResumeCache, or replay an entire
+experiment to perform additional computation such as metric using the
+ReplayCache.
+"""
+
+import shutil
+import os
+import numpy as np
+import dataset
+
+from sklearn.datasets import load_iris
+from sklearn.svm import SVC
+from sklearn.model_selection import train_test_split
+
+from cardinal.random import RandomSampler
+from cardinal.uncertainty import MarginSampler
+from cardinal.cache import ReplayCache
+from cardinal.utils import GrowingIndex
+
+##############################################################################
+# Since we will be looking at the cache, we need a utility function to display
+# a tree folder.
+
+def print_folder_tree(startpath):
+    for root, dirs, files in os.walk(startpath):
+        level = root.replace(startpath, '').count(os.sep)
+        indent = ' ' * 4 * (level)
+        print('{}{}/'.format(indent, os.path.basename(root)))
+        subindent = ' ' * 4 * (level + 1)
+        for f in files:
+            print('{}{}'.format(subindent, f))
+
+#############################################################################
+# We load the data and define the parameters of this experiment:  
+#
+# * ``batch_size`` is the number of samples that will be annotated and added to
+#   the training set at each iteration,
+# * ``n_iter`` is the number of iterations in our simulation
+
+iris = load_iris()
+X = iris.data
+y = iris.target
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.5, random_state=1)
+batch_size = 5
+n_iter = 10
+
+model = SVC(probability=True)
+
+samplers = [
+    ('margin', MarginSampler(model, batch_size))
+]
+
+#############################################################################
+# We define our experiment in a dedicated function since we want to run it
+# several times. We also create a dedicated exception that we will rise to
+# simulate an interruption in the experiment.
+#
+# Note the use of the GrowingIndex utils that facilitate the handing of
+# indices in an active learning experiment.
+
+
+class ExampleError(Exception):
+    pass
+
+
+def run(force_failure=False, compute_metric=None):
+
+    for sampler_name, sampler in samplers:
+
+        config = dict(sampler=sampler_name)
+
+        with ReplayCache('./cache', './cache.db', keys=config) as cache:
+
+            index = GrowingIndex(X_train.shape[0])
+
+            # Add at least one sample from each class
+            index.add_to_selected([np.where(y_train == i)[0][0] for i in np.unique(y)])
+
+            selected = cache.variable('selected', index.selected)
+
+            for j, prev_selected in cache.iter(range(n_iter), selected.previous()):
+                print('Computing iteration {}'.format(j))
+                index.resume(prev_selected)
```

Comment:
I've only read the client code here but it's not clear or at least not easy to guess what this does ?

Reply:
Yes for now it's not clear because:
- I have this "GrowingIndex" (terrible name too) that is convenient to perform AL experiments
- I have variables that allow to track values and resume experiments.

My problem here is that I want to have a mix of both. One option is to modify variable to handle more complex types, such as this index. The other option is to allow using a tracked varaible inside the index. I have not found the best option yet.